### ISSUE TITLE
Breadcrumbs do not preserve query parameters

### ISSUE THREAD

**Issue by jonest** [HealthCatalyst/Fabric.Cashmere#740]
**Describe the bug**
We're trying to keep track of some state using query parameters, but the breadcrumbs don't seem to work with them.

I've tried messing around with the library, but nothing seems to work correctly. Hoping someone else has some thoughts

![image](https://user-images.githubusercontent.com/3075414/53058769-bab6c600-3471-11e9-900a-f6859100a040.png)

**Comment by benjanderson** [comment]
We knew after the breadcrumbs component it would have quite a few shortcomings as there are many scenarios it doesn't cover (see #125). Instead of trying to accommodate for all of them, we decided to abstract the styles and give a guide to use them. It is not a very complex component, so it seemed like the best solution without spending a bunch of time on a super robust solution.

https://cashmere.healthcatalyst.net/styles/breadcrumbs

Before I found the style guide I was also making a demo

https://stackblitz.com/edit/cashmere-demo-breadcrumbs
